OpenRA.Mods.Common/Warheads/CreateResourceWarhead.cs

```diff
@@ -46,13 +49,17 @@ public override void DoImpact(Target target, WarheadArgs args)
 				Log.Write("debug", "Warhead defines an invalid resource type '{0}'".F(AddsResourceType));
```

Comment:
We can both simplify things here and make this more correct/useful for modders by checking for this in
```
void IRulesetLoaded<WeaponInfo>.RulesetLoaded(Ruleset rules, WeaponInfo info)
```
and throwing an exception during load/lint. See `SpreadDamageWarhead` for an example to copy.